base/reflection.jl

```diff
@@ -1258,10 +1258,11 @@ function code_typed_opaque_closure(@nospecialize(closure::Core.OpaqueClosure);
         debuginfo::Symbol=:default,
         interp = Core.Compiler.NativeInterpreter(closure.world))
     ccall(:jl_is_in_pure_context, Bool, ()) && error("code reflection cannot be used from generated functions")
-    if isa(closure.source, Method)
-        code = _uncompressed_ir(closure.source, closure.source.source)
+    m = closure.source
+    if isa(m, Method)
+        code = _uncompressed_ir(m, m.source)
         debuginfo === :none && remove_linenums!(code)
-        return Any[Pair{CodeInfo,Any}(code, code.rettype)]
```

Comment:
This change seems very unlikely to be correct. Could perhaps be `::Type`, but `::Any` seems clearer to me.

Reply:
I think I basically wanted to make the printing consistent with `code_typed_by_type`:
```julia
julia> import Base.Experimental: @opaque

julia> oc = @opaque x -> sin(x)
(::Any)::Any->◌

julia> code_typed(oc, (Int,))
1-element Vector{Any}:
 Pair{Core.CodeInfo, Any}(CodeInfo(
    @ none:1 within `opaque closure`
1 ─ %1 = Main.sin(x)
└──      return %1
), Any)

julia> code_typed(x->sin(x), (Any,))
1-element Vector{Any}:
 CodeInfo(
1 ─ %1 = Main.sin(x)::Any
└──      return %1
) => Any
```

But yeah, this is not that correct. Maybe we can just resort to dynamic dispatch here (since this is not really performance sensitive).